test/datacollector.UnitTests/DummyDataCollectors.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.DataCollector.UnitTests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Xml;
+
+    using Microsoft.VisualStudio.TestPlatform.DataCollector.Interfaces;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel.DataCollection;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel.DataCollector;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel.Logging;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+
+    [DataCollectorFriendlyName("CustomDataCollector")]
+    [DataCollectorTypeUri("my://custom/datacollector")]
+    public class CustomDataCollector : DataCollector, ITestExecutionEnvironmentSpecifier
+    {
+        public static bool IsInitialized = false;
+        public static bool IsSessionStartedInvoked = false;
+        public static bool IsSessionEndedInvoked = false;
+        public static bool GetTestExecutionEnvironmentVariablesThrowException;
+        public static IEnumerable<KeyValuePair<string, string>> EnvVarList;
```

Comment:
As we are loading the datacollectors through run settings, it is not possible to dynamically construct the object and load it through run settings.